samples/pmic/npm6001/src/main.c

```diff
@@ -0,0 +1,530 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+
+#include <dk_buttons_and_leds.h>
+
+#include <lib_npm6001.h>
+
+LOG_MODULE_REGISTER(main, CONFIG_MAIN_LOG_LEVEL);
+
+#define LED_INIT_OK DK_LED1
+#define LED_BUCK3_ON DK_LED2
+#define LED_GPIO_ON DK_LED4
+
+#define BUCK0_DEFAULT_VOLTAGE 3300
+#define BUCK1_DEFAULT_VOLTAGE 1200
+#define BUCK2_DEFAULT_VOLTAGE 1400
+#define BUCK3_DEFAULT_VOLTAGE 500
+#define LDO0_DEFAULT_VOLTAGE 2700
+
+K_SEM_DEFINE(n_int_sem, 0, 1);
+
+static const struct device *i2c_dev = DEVICE_DT_GET_ONE(nordic_nrf_twim);
```

Comment:
this is fragile: you need to capture to which I2C you're connecting the PMIC in DT. PMIC needs to be described in Devicetree somehow.